tests/framework/statistics/criteria.py

```diff
@@ -26,65 +26,120 @@ def msg(self, msg):
         self._msg = msg
 
 
+# pylint: disable=R0903
+class CriteriaFactory:
+    """Comparison criteria factory class."""
+
+    @classmethod
+    def get(cls, criteria_cls_name) -> 'ComparisonCriteria':
+        """`criteria_cls_name` must be a valid criteria class name."""
+        return locate(f"framework.statistics.criteria.{criteria_cls_name}")
+
+
 # pylint: disable=R0903
 class ComparisonCriteria(ABC):
     """Comparison criteria between results and targets."""
 
-    def __init__(self, name: str, target: Number):
-        """Initialize the comparison criteria."""
-        self.target = target
-        self.name = name
+    def __init__(self, name: str, baseline: dict):
+        """Initialize the comparison criteria.
+
+        Baseline expected schema:
+        ```
+        {
+            "type": "object",
+            "properties": {
+                "target": number,
+            },
+            "required": ["target"]
+        }
+        ```
+        """
+        self._baseline = baseline
+        self._name = name
 
     @abstractmethod
     def check(self, actual):
         """Compare the target and the actual."""
 
+    @property
+    def name(self):
+        """Return criteria name."""
+        return self._name
+
+    @property
+    def target(self):
+        """Return criteria target."""
+        target = self._baseline.get("target")
+        if not target:
+            raise Failed("Baseline target not defined.")
+
+        return target
+
 
 # pylint: disable=R0903
-class GraterThan(ComparisonCriteria):
+class GreaterThan(ComparisonCriteria):
     """Greater than comparison criteria."""
 
-    def __init__(self, target: Number):
+    def __init__(self, baseline: dict):
         """Initialize the criteria."""
-        super().__init__("GreaterThan", target)
+        super().__init__("GreaterThan", baseline)
 
     def check(self, actual):
         """Compare the target and the actual."""
         fail_msg = self.name + f" failed. Target: '{self.target} " \
                                f"vs Actual: '{actual}'."
-        if self.target > actual:
+        if actual < self.target:
             raise Failed(msg=fail_msg)
 
 
 # pylint: disable=R0903
 class LowerThan(ComparisonCriteria):
     """Lower than comparison criteria."""
 
-    def __init__(self, target: Number):
+    def __init__(self, baseline: dict):
         """Initialize the criteria."""
-        super().__init__("LowerThan", target)
+        super().__init__("LowerThan", baseline)
 
     def check(self, actual):
         """Compare the target and the actual."""
         fail_msg = self.name + f" failed. Target: '{self.target} " \
```

Comment:
I'd say to also have this `fail_msg` as a property method on the base class, which can be overriden on a case-by-case basis (for the `EqualWith` operator), to minimise duplication

Reply:
Done.